universal-application-tool-0.0.1/app/services/question/types/QuestionDefinition.java

```diff
@@ -253,6 +254,42 @@ public ImmutableSet<CiviFormError> validate() {
       if (numUniqueOptionDefaultValues != multiOptionQuestionDefinition.getOptions().size()) {
         errors.add(CiviFormError.of("Multi-option question options must be unique"));
       }
+      int numOptions = multiOptionQuestionDefinition.getOptions().size();
+      OptionalInt minChoicesRequired =
+          multiOptionQuestionDefinition.getMultiOptionValidationPredicates().minChoicesRequired();
+      OptionalInt maxChoicesAllowed =
+          multiOptionQuestionDefinition.getMultiOptionValidationPredicates().maxChoicesAllowed();
+      if (minChoicesRequired.isPresent()) {
+        if (minChoicesRequired.getAsInt() < 0) {
+          errors.add(CiviFormError.of("Minimum number of choices required cannot be negative"));
+        }
+        if (minChoicesRequired.getAsInt() > numOptions) {
+          errors.add(
+              CiviFormError.of(
+                  "Minimum number of choices required cannot exceed the number of options"));
+        }
+      }
+      if (maxChoicesAllowed.isPresent()) {
+        if (maxChoicesAllowed.getAsInt() < 0) {
+          errors.add(CiviFormError.of("Maximum number of choices allowed cannot be negative"));
+        }
+        if (maxChoicesAllowed.getAsInt() > numOptions) {
+          errors.add(
+              CiviFormError.of(
+                  "Maximum number of choices allowed cannot exceed the number of options"));
+        }
+      }
+      if (minChoicesRequired.isPresent() && maxChoicesAllowed.isPresent()) {
+        if (minChoicesRequired.getAsInt() == 0 && maxChoicesAllowed.getAsInt() == 0) {
+          errors.add(CiviFormError.of("Cannot require exactly 0 choices"));
+        }
+        if (minChoicesRequired.getAsInt() > maxChoicesAllowed.getAsInt()) {
+          errors.add(
+              CiviFormError.of(
+                  "Minimum number of choices required must be less than or equal to the maximum"
+                      + " choices allowed"));
+        }
+      }
```

Comment:
nit: some additional whitespace in here would make it a little easier for my eyes -- blank lines between multi-line statements (e.g. `if` statements) and other code at the same level of indentation helps a lot imo

Reply:
Done!